client/src/GamemasterInGame/PlayerListComponents/PlayerNoteInput.tsx

```diff
@@ -7,16 +7,25 @@ interface PlayerNoteInputProps {
   player: string;
   children: React.ReactNode;
   note?: string;
+  saveNote?: (newNote: string) => void;
+  hideRole?: boolean;
 }
 
 export function PlayerNoteInput({
   player,
   children,
   note = "",
+  saveNote,
+  hideRole = false,
```

Comment:
This component's getting pretty unfocused and should probably be split for the two use cases.